rcl/test/rcl/test_time.cpp

```diff
@@ -205,6 +205,29 @@ TEST_F(CLASSNAME(TestTimeFixture, RMW_IMPLEMENTATION), test_ros_clock_initially_
   EXPECT_EQ(0, query_now);
 }
 
+TEST(CLASSNAME(rcl_time, RMW_IMPLEMENTATION), time_point_validation) {
+  ASSERT_FALSE(rcl_time_point_valid(NULL));
+
+  rcl_time_point_t uninitialized;
+  (void) uninitialized;
+  // Not reliably detectable due to random values.
+  // ASSERT_FALSE(rcl_clock_valid(&uninitialized));
```

Comment:
As we talked about in person @methylDragon, we should just not have stuff in the tests like this (I know you're copying from the clock tests, but I consider them to be not ideal as well) and instead we should document that using uninitialized values with any of our functions is undefined behavior, e.g.:

https://github.com/ros2/rcl/blob/e47ed58d8d2018ee0fdb309b130bffa1e18c0c3d/rcl/include/rcl/node.h#L96-L99

Node is just an example, we should ideally document that all structures have to be zero initialized before being used (unless we have specific cases where we want to avoid that for performance reasons).

I don't see the `rcl_..._zero_initialized_...()` like functions for time/clock objects. We should probably clean them up to have these, again unless there are specific scenarios where we want to avoid that for performance reasons. Like it might be ok to use an uninitialized time point if passing it to get time (which is write only maybe). Avoiding an unnecessary initialization state. But functions that read the struct, like "is valid" should definitely be documented to not work with uninitialized values. In which case we don't need commented blocks like these in the tests.